src/components/10-atoms/radio/index.js

```diff
@@ -0,0 +1,251 @@
+    this.nogap = false;
+    this.isReact = false;
+    this.onFocus = () => {};
+    this.onBlur = () => {};
+    this.onChange = () => {};
+  }
+
+  getDefaultName() {
+    this.name = this.name || defaultName();
+    return this.name;
+  }
+
+  render() {
+    const {
+      id,
+      name = this.getDefaultName(),
+      button,
+      icon,
+      checked,
+      value,
+      disabled,
+      label,
+      isReact,
+      state: { isControlled },
+    } = this;
+
+    // controlledness is a React-only notion
+    this.state.isControlled = isControlled && isReact;
+
+    return html`
+      <label class="a-radio__wrapper">
+        <input
+          id="${id}"
+          class="a-radio__input"
+          type="radio"
+          name="${name}"
+          ?checked="${checked}"
+          value="${value}"
+          ?disabled="${disabled}"
+          @change="${this.handleChange}"
+          @focus="${this.onFocus}"
+          @blur="${this.onBlur}"
+        />
+        ${button
+          ? html``
+          : html`
+              <span class="a-radio__icon"></span>
+            `}
+        ${icon ? svg([icon]) : html``}
+        <div class="a-radio__content">${label}</div>
+      </label>
+    `;
+  }
+
+  firstUpdated() {
+    this.input = this.querySelector('input');
+    const { name, button, noautowidth } = this;
+    const ourButton = this.querySelector('.a-radio__content');
+    radioButtonGroup[name] = radioButtonGroup[name] || new Set();
+    radioButtonGroup[name].add(ourButton);
+
+    if (button) {
+      setTimeout(() => {
+        const { width: labelTextWidth } = this.querySelector(
+          '.a-radio__content'
+        ).getBoundingClientRect();
+        maxWidth[name] = Math.max(labelTextWidth | 0, maxWidth[name] | 0);
+        // equalize width for all <axa-radio button> with same name:
+        const width = noautowidth ? labelTextWidth : maxWidth[name];
+        radioButtonGroup[name].forEach(radioButton => {
+          // special case 'noautowidth' only impose minWidth on ourselves
+          // (suppresses length changes .a.k.a 'punping effect' between
+          // selected/unselected state due to font-weight changes)
+          if (noautowidth && radioButton !== ourButton) return;
+          radioButton.style.minWidth = `${width + WIDTH_SLACK}px`;
+        });
+      }, /* give DOM some time to paint before measuring width */ DOM_PAINT_TIME);
+    }
+  }
+
+  updated() {
+    const {
+      name,
+      input,
+      state: { isControlled, checked },
+    } = this;
+
+    // we are the selected element of a group of controlled inputs?
+    if (isControlled && checked && name) {
+      // yes, perform forced re-select of radio button
+      // to maintain controlledness in the face of native UI-state changes
+      input.checked = checked;
+    }
+
+    // register ourselves as currently selected in same-named group of all
+    // radio buttons
+    if (this.checked) selectedRadioButton[name] = this;
```

Comment:
please don't relay on row line for scope, use brackets always.

Reply:
sometimes it just looks better, but: done.